src/code/z_effect_soft_sprite_old_init.c

```diff
@@ -214,255 +328,718 @@ void EffectSsBomb2_SpawnLayered(GlobalContext* globalCtx, Vec3f* pos, Vec3f* vel
+    initParams.scale = scale;
+
+    EffectSs_Spawn(globalCtx, EFFECT_SS_G_MAGMA2, 128, &initParams);
+}
 
 // EffectSsStone1 Spawn Functions
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_effect_soft_sprite_old_init/func_80029C00.s")
+void EffectSsStone1_Spawn(GlobalContext* globalCtx, Vec3f* pos, s32 arg2) {
+    EffectSsStone1InitParams initParams;
+
+    initParams.pos = *pos;
+    initParams.unk_C = arg2;
+
+    EffectSs_Spawn(globalCtx, EFFECT_SS_STONE1, 128, &initParams);
+}
 
 // EffectSsHitMark Spawn Functions
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_effect_soft_sprite_old_init/func_80029C50.s")
+void EffectSsHitMark_Spawn(GlobalContext* globalCtx, s32 type, s16 scale, Vec3f* pos) {
+    EffectSsHitMarkInitParams initParams;
+
+    initParams.type = type;
+    initParams.scale = scale;
+    Math_Vec3f_Copy(&initParams.pos, pos);
+
+    EffectSs_Spawn(globalCtx, EFFECT_SS_HITMARK, 128, &initParams);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_effect_soft_sprite_old_init/func_80029CA4.s")
+void EffectSsHitMark_SpawnFixedScale(GlobalContext* globalCtx, s32 type, Vec3f* pos) {
+    EffectSsHitMark_Spawn(globalCtx, type, 300, pos);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_effect_soft_sprite_old_init/func_80029CC8.s")
+void EffectSsHitMark_SpawnCustomScale(GlobalContext* globalCtx, s32 type, s16 scale, Vec3f* pos) {
+    EffectSsHitMark_Spawn(globalCtx, type, scale, pos);
+}
 
-void EffectSsFhgFlash_Spawn(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 arg4, u8 arg5) {
+// EffectSsFhgFlash Spawn Functions
+
+/**
+ * Spawn a light ball effect
+ *
+ * param changes the color of the ball. Refer to FhgFlashLightBallParam for the options.
+ * Note: this type requires OBJECT_FHG to be loaded
+ */
+void EffectSsFhgFlash_SpawnLightBall(GlobalContext* globalCtx, Vec3f* pos, Vec3f* velocity, Vec3f* accel, s16 scale,
+                                     u8 param) {
     EffectSsFhgFlashInitParams initParams;
 
     Math_Vec3f_Copy(&initParams.pos, pos);
     Math_Vec3f_Copy(&initParams.velocity, velocity);
     Math_Vec3f_Copy(&initParams.accel, accel);
-    initParams.unk_24 = arg4;
-    initParams.unk_26 = arg5;
-    initParams.unk_2C = 0;
+    initParams.scale = scale;
+    initParams.param = param;
+    initParams.type = FHGFLASH_LIGHTBALL;
 
     EffectSs_Spawn(globalCtx, EFFECT_SS_FHG_FLASH, 128, &initParams);
 }
 
-void EffectSsFhgFlash_Spawn2(GlobalContext* globalCtx, Actor* arg1, Vec3f* pos, s16 arg3, u8 arg4) {
+/**
+ * Spawn a lightning shock effect
+ *
+ * param determines where the ligntning should go
+ * 0: dont attach to any actor. spawns at the position specified by pos
+ * 1: spawn at one of Player's body parts, chosen at random
+ * 2: spawn at one of Phantom Ganon's body parts, chosen at random
+ */
+void EffectSsFhgFlash_SpawnLightning(GlobalContext* globalCtx, Actor* actor, Vec3f* pos, s16 scale, u8 param) {
     EffectSsFhgFlashInitParams initParams;
```

Comment:
note to self, gonna rename this EffectSsFhgFlash_SpawnShock later (and related param names)